netsync/manager.go

```diff
@@ -1070,9 +1070,6 @@ func (sm *SyncManager) handleBlockMsgHeadersFirst(bmsg *blockMsg) {
 	// Add the block to the queue.
 	sm.queuedBlocks[*blockHash] = bmsg
 
-	// Remove block from the request map as we've added it to queuedBlocks.
-	delete(state.requestedBlocks, *blockHash)
```

Comment:
Removing this check creates a DoS vector as the peer can now keep sending us orphan blocks to fill up the memory used by `b.orphans`.